### ISSUE TITLE
Redefined YAML subblock in ./rpminspect.yaml overrides whole block in user-defined config.yaml, not just a subblock.

### ISSUE THREAD

**Issue by zmiklank** [rpminspect/rpminspect#576]
### Example:
User-specified `rpminspect.yaml` contains following YAML section:
```
badfuncs:
    - gethostbyname
    - gethostbyaddr
```
The local `./rpminspect.yaml` wants to ignore check of these functions in `somefile`, thus contains:
```
badfuncs:
  ignore:
    /path/to/somefile
```
Based on [1] I concluded that when combining above mentioned conf files, the result will be: 
    "check the RPM for these functions, but do not check `somefile`":
```
badfuncs:
    - gethostbyname
    - gethostbyaddr
  ignore:
    /path/to/somefile
``` 

However, based on `--dump-config` the actual result is:
    "no badfuncs are specified and do not check for them in `somefile`":
```
badfuncs:
  ignore:
    /path/to/somefile
```

This freeing of `bad_functions` is in `lib/init.c`:
```
 723                         if (group != BLOCK_BADFUNCS) {
 724                             list_free(ri->bad_functions, free);
 725                             ri->bad_functions = NULL;
 726                             group = BLOCK_BADFUNCS;
 727                         }
```
I guess, that this freeing is needed when the `./rpminspect.yaml` wants to set its own forbidden functions, and does not want to use the ones from the user-specified conf file.

What is in this case expected behavior, please? Am I misunderstanding the situation?

[1] https://rpminspect.readthedocs.io/en/latest/configuration.html#rpminspect-yaml

**Comment by dcantrell** [comment]
The user-defined rpminspect.yaml file is supposed to override entire blocks.  But, if you want to redefine the ignore list for badfuncs, you do not also need to redefine the list of functions.  The overrides happen only if you define a block or subblock in your rpminspect.yaml file.

In the example you provided, rpminspect should have redefined the list of functions badfuncs checks for to gethostbyname and gethostbyaddr and the ignore list for badfuncs should only contain /path/to/somefile.  Is that what you are trying to define?  If so, you may have found a bug.  I just need to understand how to reproduce it locally to fix it.

**Comment by zmiklank** [comment]
Yes, I wanted to override only the `ignore` subblock in `badfuncs` block, so that for the exemplary functions `gethostbyname` and `gethostbyaddr` would not be checked in the `somefile` file, but the specified `ignore` subblock of `badfuncs` YAML block overrode whole `badfuncs` YAML block..

**Comment by zmiklank** [comment]
I would say that PR https://github.com/rpminspect/rpminspect/pull/693 fixes this issue. Hence I am closing it, thank you!
